File: object/bookfullinfo.cpp
#include "bookfullinfo.h"

BookFullInfo::BookFullInfo(QString cover,
                           QString url,
                           QString title,
                           QString authors,
                           QString rating,
                           QString description,
                           QString categories,
                           QString edition,
                           QString language,
                           QString isbn,
                           QString isbn10,
                           QString isbn13,
                           QString file,
                           QString volume,
                           QString year,
                           QString publisher,
                           QString pages,
                           QString series,
                           QString ipfs)
    : cover(cover)
    , url(url)
    , title(title)
    , authors(authors)
    , rating(rating)
    , description(description)
    , categories(categories)
    , edition(edition)
    , language(language)
    , isbn(isbn)
    , isbn10(isbn10)
    , isbn13(isbn13)
    , file(file)
    , volume(volume)
    , year(year)
    , publisher(publisher)
    , pages(pages)
    , series(series)
    , ipfs(ipfs)
{

}

BookFullInfo::~BookFullInfo()
{

}

File: book_full_info.py
from dataclasses import dataclass, asdict
import json

@dataclass
class BookFullInfo:
    cover: str
    url: str
    title: str
    authors: str
    rating: str
    description: str
    categories: str
    edition: str
    language: str
    isbn: str
    isbn10: str
    isbn13: str
    file: str
    volume: str
    year: str
    publisher: str
    pages: str
    series: str
    ipfs: str

    def transform_2json_str(self) -> str:
        return json.dumps(asdict(self))